js/search.js

```diff
@@ -0,0 +1,416 @@
+"use strict";
+(function () {
+  const searchinfo = location.search.substr(1).split("_");
+
+  if (
+    searchinfo[0].split("-")[0] !== "text" ||
+    searchinfo[1] === undefined ||
+    searchinfo[1].split("=")[1] === undefined
```

Comment:
same
just use if (obj) or if (!obj)
empty strin, 0, null, undefined will be considered as false